src/identity_tree.rs

```diff
@@ -673,6 +675,50 @@ impl CanonicalTreeBuilder {
         builder
     }
 
+    pub fn restore(
+        tree_depth: usize,
+        dense_prefix_depth: usize,
+        initial_leaf: &Field,
+        last_index: usize,
+        leftover_items: &[ruint::Uint<256, 4>],
+        flattening_threshold: usize,
+        mmap_file_path: &str,
+    ) -> Option<Self> {
+        let tree: LazyMerkleTree<PoseidonHash, lazy_merkle_tree::Canonical> =
+            match PoseidonTree::<lazy_merkle_tree::Canonical>::attempt_dense_mmap_restore(
+                tree_depth,
+                dense_prefix_depth,
+                initial_leaf,
+                mmap_file_path,
+            ) {
+                Ok(tree) => tree,
+                Err(error) => {
+                    warn!("Tree wasn't restored. Reason: {}", error.to_string());
+                    return None;
+                }
+            };
```

Comment:
What happens if the tree restores from the file but the values are actually different than what we expect?

I don't think this can currently happen, because at startup we only mark as "processed"not "mined", but technically such a situation would be possible if the sequencer crashed after submitting a deletion batch and then restarted with the updated values (if we ever start marking as "mined" at startup).